core/frontend/src/main/scala/io/udash/routing/UrlChangeProvider.scala

```diff
@@ -5,13 +5,16 @@ import io.udash.core.Url
 import io.udash.properties.MutableBufferRegistration
 import io.udash.utils.Registration
 import org.scalajs.dom
-import org.scalajs.dom.Element
+import org.scalajs.dom.{Element, Location}
 import org.scalajs.dom.raw.HashChangeEvent
 
 import scala.scalajs.js
 
 /** Provides information about current URL. */
 trait UrlChangeProvider {
```

Comment:
Is there a reason not to test both of these providers?

Reply:
It's hard to write any useful unit tests for this. I'm going to move selenium tests from the guide to the core project in the nearest future - I think that's the way to test `UrlChangeProvider`s.